iOS/UZTickets/src/scan/UZTScanViewController.m

```diff
@@ -0,0 +1,180 @@
+//
+//  UZTScanViewController.m
+//  UZTickets
+//
+//  Created by nekoi on 6/1/15.
+//  Copyright (c) 2015 DAloG ltd. All rights reserved.
+//
+
+#import "UZTScanViewController.h"
+#import "UZTScanView.h"
+
+@interface UZTScanViewController ()<AVCaptureMetadataOutputObjectsDelegate>
+
+@property UZTScanView* view;
+@property (strong) AVCaptureMetadataOutput* output;
+@property (strong) AVCaptureSession* session;
+@property (strong) dispatch_queue_t dispatchQueue;
+
+@end
+
+@implementation UZTScanViewController
+@dynamic view;
+
++ (BOOL)canScan
+{
+    NSArray* devices = [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo];
+    if (devices.count == 0) return NO;
+    
+    AVCaptureDevice* device;
```

Comment:
This is question not of reuse but concern separation. ViewController is just a view class. And AV stuff is pretty heavy. Working with real devices not look like UI stuff... can I refactor it?